### PR TITLE
Summary.sevt

### PR DESCRIPTION
**Pull request by FedericoCarli**

### CODE REVIEW

File: R/1b-base-model-function.R [hunk 1]
```diff
@@ -673,17 +673,30 @@ summary.sevt <- function(object, ...) {
...
     out[[vns[1]]][["sample.size"]] <- attr(object$prob[[vns[1]]][[1]], "n")
+    
+    for(j in 1:length(attr(object$prob[[vns[1]]][[1]], "names"))) {
+      out[[vns[1]]][[attr(object$prob[[vns[1]]][[1]], "names")[j]]] <- as.numeric(object$prob[[vns[1]]][[1]][j])
+    }
   }
   for (i in 2:nv){
     v <- vns[i]
-    D <- data.frame(stage = unique(object$stages[[v]]))
+    D <- data.frame(stage = factor(sort(unique(as.numeric(object$stages[[v]])))))
```
Comment (gherardovarando) [inline R/1b-base-model-function.R]: what happens here if object$stages[[v]]  is not a number-character ("1", "7", "11"). stage names should work also with arbitrary strings ("stagename", "red", etc...)

Comment (FedericoCarli) [reply R/1b-base-model-function.R]: This was the first modification I did. Actually, it should work also without the re-ordering I did here and 4-5 lines below:



summary.sevt <- function(object, ...) {
  stopifnot(is(object, "sevt"))
  vns <- varnames.sevt(object)
  nv <- nvar.sevt(object)
  out <- list()
  out[[vns[1]]] <- data.frame(stage = "1", 
                              npaths = 0)
  if (is_fitted.sevt(object)){
    out[[vns[1]]][["sample.size"]] <- attr(object$prob[[vns[1]]][[1]], "n")
    
    for(j in 1:length(attr(object$prob[[vns[1]]][[1]], "names"))) {
      out[[vns[1]]][[attr(object$prob[[vns[1]]][[1]], "names")[j]]] <- as.numeric(object$prob[[vns[1]]][[1]][j])
    }
  }
  for (i in 2:nv){
    v <- vns[i]
    D <- data.frame(stage = unique(object$stages[[v]]))
    D$npaths <- vapply(D$stage, function(s){
      sum(object$stages[[v]] == s)
    }, FUN.VALUE = 1)

    if (is_fitted.sevt(object)){
      D[["sample.size"]] <- vapply(D$stage, function(s){
        attr(object$prob[[v]][[as.character(s)]], "n")
      }, FUN.VALUE = 1)
      
      for(k in 1:NROW(D)) {
        for(j in 1:length(attr(object$prob[[v]][[D$stage[k]]], "names"))) {
          D[k, attr(object$prob[[v]][[as.character(D$stage[k])]], "names")[j]] <- as.numeric(object$prob[[v]][[as.character(D$stage[k])]][j])
        }
      }
    }
    out[[vns[i]]] <- D
  }
  out <- list(stages.info = out)
  out$call <- object$call
  out$ll <- object$ll
  out$lambda <- object$lambda
  class(out) <- "summary.sevt"
  return(out)
}

Comment (gherardovarando) [reply R/1b-base-model-function.R]: Ok I will try to implement it

Comment (FedericoCarli) [reply R/1b-base-model-function.R]: I added a test for summary.sevt

**Comment by FedericoCarli** [comment]
We have that small issue. I think the summary should return sample.size and probabilities also on a subtree.

s0 <- full(PhDArticles)
summary(s0) # it returns NaN probability distributions when sample.size == 0. Is it ok for you? For me it is reasonable.
s1 <- subtree.sevt(s0, "0")
summary(s1) # it does not return sample.size and probability distributions, because:
is_fitted.sevt(s0)
# TRUE
is_fitted.sevt(s1)
# FALSE

# because in the subtree we haven't the $ctables and in summary.sevt we return sample.size and prob only if(is_fitted.sevt(object)).
# is_fitted.sevt
# function(x) {
#   return(!is.null(x$prob) && !is.null(x$ctables))
# }

**Comment by gherardovarando** [comment]
Right this is rather an issue of `subtree.sevt` I will try to fix it !

**Comment by gherardovarando** [comment]
should work now

**Pull request closed without merging**